rig/machine_control/consts.py

```diff
@@ -304,3 +304,12 @@ class BMPInfoType(enum.IntEnum):
     can_status = 2  # Status of all CAN devices on the bus
     adc = 3  # ADC (e.g. voltage + temperature)
     ip_addr = 4  # IP Address
+
+
+# Dictionary of (address % 4, n_bytes % 4) to data type
+address_length_dtype = {
```

Comment:
I haven't benchmarked this. I'll give that a go later. My only thoughts are that Python's hashing is known awesome.